client/consensus/manual-seal/src/consensus/babe.rs

```diff
@@ -122,14 +126,45 @@ impl<B, C> ConsensusDataProvider<B> for BabeConsensusDataProvider<B, C>
 			})?;
 
 		// this is a dev node environment, we should always be able to claim a slot.
-		let (predigest, _) = authorship::claim_slot(slot_number, epoch.as_ref(), &self.keystore)
-			.ok_or_else(|| Error::StringError("failed to claim slot for authorship".into()))?;
-
-		Ok(Digest {
-			logs: vec![
+		let logs =  if let Some((predigest, _)) = authorship::claim_slot(slot_number, epoch.as_ref(), &self.keystore) {
+			vec![
 				<DigestItemFor<B> as CompatibleDigestItem>::babe_pre_digest(predigest),
-			],
-		})
+			]
+		} else {
+			drop(epoch);
```

Comment:
Why do we explicitly drop it?

Reply:
the epoch holds a ref to shared_epoch_changes, which prevents us from acquiring a mutable reference to epoch in line 146